selectel/internal/api/servers/servers.models.go

```diff
@@ -0,0 +1,184 @@
+package servers
+
+import (
+	"net"
+	"slices"
+	"strings"
+)
+
+type (
+	Server struct {
+		ID                 string             `json:"uuid"`
+		Name               string             `json:"name"`
+		Available          []*ServerAvailable `json:"available"`
+		PricePlanAvailable []string           `json:"price_plan_available"`
+		Tags               []string           `json:"tags"`
+
+		IsServerChip bool
+	}
+
+	ServerAvailable struct {
+		LocationID string                      `json:"location"`
+		PlanCount  []*ServerAvailablePricePlan `json:"plan_count"`
+	}
+
+	ServerAvailablePricePlan struct {
+		Count    int    `json:"count"`
+		PlanUUID string `json:"plan_uuid"`
+	}
+
+	Servers []*Server
+)
+
+func (s Server) IsLocationAvailable(locationID string) bool {
+	for _, available := range s.Available {
+		if available.LocationID == locationID {
+			return true
+		}
+	}
+
+	return false
+}
+
+func (s Server) IsPrivateNetworkAvailable() bool {
+	return !s.IsServerChip &&
+		!slices.ContainsFunc(s.Tags, func(s string) bool {
+			return s == "10GE_Internet_MC-LAG" || s == "10GE_Internet" || s == "25GE_Local_MC-LAG" || s == "10GE_Local" ||
+				s == "10GE_Local_MC-LAG"
+		})
+}
+
+func (s Server) IsPricePlanAvailableForLocation(pricePlanID, locationID string) bool {
+	var foundAvailable bool
+	for _, psAvailable := range s.PricePlanAvailable {
+		if psAvailable == pricePlanID {
+			foundAvailable = true
+			break
+		}
+	}
+
+	if !foundAvailable {
+		return false
+	}
+
+	for _, avByLoc := range s.Available {
+		if avByLoc.LocationID == locationID {
+			for _, planCount := range avByLoc.PlanCount {
+				if planCount.PlanUUID == pricePlanID && planCount.Count >= 1 {
+					return true
+				}
+			}
+		}
+	}
```

Comment:
Optional: nesting can be reduced by invert condition
```suggestion
	for _, avByLoc := range s.Available {
		if avByLoc.LocationID != locationID {
			continue
		}
		for _, planCount := range avByLoc.PlanCount {
			if planCount.PlanUUID == pricePlanID && planCount.Count >= 1 {
				return true
			}
		}
	}

```

Reply:
I agree, that's better. Added these changes